File: ch4/prime/M868/main.go
// 试除法---质数判定
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	r := bufio.NewReader(os.Stdin)
	n, a := 0, 0
	fmt.Fscan(r, &n)

	for ; n != 0; n-- {
		fmt.Fscan(r, &a)
		if isPrime(a) {
			fmt.Println("Yes")
		} else {
			fmt.Println("No")
		}
	}
}

func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	for i := 2; i <= n/i; i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

File: ch4/prime/M868/main.py
# math
import math


def is_prime(n):
    if n < 2:
        return False

    end = int(math.sqrt(n)) + 1
    for i in range(2, end):
        if n % i == 0:
            return False

    return True


n = int(input())

while n:
    n -= 1
    x = int(input())
    if is_prime(x):
        print("Yes")
    else:
        print("No")
